codalab/worker/singularity_image_manager.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import os
+import threading
+
+from codalab.lib.formatting import size_str
+from codalab.worker.image_manager import ImageAvailabilityState
+from codalab.worker.fsm import DependencyStage
+from codalab.worker.image_manager import ImageManager
+from codalab.worker.singularity_utils import get_singularity_container_size
+from spython.main import Client  # type: ignore
```

Comment:
Add spython to https://github.com/codalab/codalab-worksheets/blob/master/setup.cfg instead of using "type: ignore"